docs/content/manual/misc/3902-elemental-cloud-init-harvester-crds.md

```diff
@@ -0,0 +1,217 @@
+  2. the contents of `99_filewrite.yaml` look appropriate
+3. Validate that `kubectl describe cloudinits/write-file-with-non-suffix-filename` looks appropriate
+4. Delete the CloudInit CRD via `kubectl delete cloudinits/write-file-with-non-suffix-filename`
+
+#### Validate That Non-YAML CloudInit CRD is Rejected
+1. Prepare a YAML loadout of a CloudInit resource that takes the shape of:
+```yaml
+apiVersion: node.harvesterhci.io/v1beta1
+kind: CloudInit
+metadata:
+  name: consolelogoverwrite
+spec:
+  matchSelector: {}
+  filename: install/console.log
+  contents: |
+    hello  there
+```
+2. When trying to apply the YAML, it should be rejected at a Webhook level, test via: `kubectl create -f your-file-name.yaml`
+3. Validate that is rejected.
+
+### Positive Tests:
+
+#### Validate that a CloudInit Resource Gets Applied Cluster Wide
+1. Prepare a YAML loadout of a CloudInit resource that takes the shape of:
+```yaml
+apiVersion: node.harvesterhci.io/v1beta1
+kind: CloudInit
+metadata:
+  name: ssh-access-test
+spec:
+  matchSelector: {}
+  filename: 99_ssh_test_cluster_wide.yaml
+  contents: |
+    stages:
+      network:
+        - authorized_keys:
+            rancher:
+              - ATTACH-YOUR-SSH-KEY-HERE
+```
+2. Apply it to the cluster via `kubectl create -f filename.yaml`
+3. Validate that the CloudInit Resource was scaffolded appropriately `kubectl describe cloudinits/ssh-access-test`
+4. Reboot all nodes in the cluster
+5. Once nodes are rebooted validate that you can log in to each node via the SSH Key provided.
+6. Delete the CRD `kubectl delete cloudinits/ssh-access-test`
+7. Remove the ssh-key entry from `/home/rancher/.ssh/authorized_keys` on each node
+8. Reboot the nodes
+9. Ensure that the `/oem/99_ssh_test_cluster_wide.yaml` is not present on any node
+
+#### Validate that a CloudInit Resource that targets a Single Node Is Applied Correctly
+1. Prepare a YAML loadout of a CloudInit resource that takes the shape of:
+```yaml
+apiVersion: node.harvesterhci.io/v1beta1
+kind: CloudInit
+metadata:
+  name: ssh-access-test-one-node
+spec:
+  matchSelector:
+    kubernetes.io/hostname: NOTE-FOR-THIS-FIELD change to be the Node Name like for instance-> harvester-node-0
+  filename: 99_ssh_test_single_node.yaml
+  contents: |
+    stages:
+      network:
+        - authorized_keys:
+            rancher:
+              - ATTACH-YOUR-SSH-KEY-HERE
+```
+2. Apply it to the cluster via `kubectl create -f filename.yaml`
+3. validate that the CloudInit resource was built appropriately `kubectl describe cloudinits/ssh-access-test-one-node`
+4. Reboot the node that has the host-name filed in, as in the node tied to `kubernetes.io/hostname`
+5. Validate that you can ssh as rancher onto the node that was rebooted without the need for password
+6. Delete the CRD `kubectl delete cloudinits/ssh-access-test-one-node`
+7. Remove the ssh-key from `/home/rancher/.ssh/authroized_keys`
```

Comment:
typo, should be `authorized`.

Reply:
Thanks for this catch :smile: :+1: - updated & force pushed fixing